lib/Net/Async/Blockchain/ETH.pm

```diff
@@ -90,29 +89,48 @@ returns a Future, the on_done response will be the accounts array.
 
 =cut
 
-async sub accounts {
+sub accounts {
     my $self = shift;
-    return $self->{accounts} //= do {
-        await $self->rpc_client->accounts();
-    };
+    return $self->{accounts};
 }
 
-=head2 update_accounts
+=head2 latest_accounts_update
 
-update the C<accounts> variable every 10 seconds
+stores the time of the latest account update
 
 =over 4
 
 =back
 
+Returns L<time> of the latest account update
+
 =cut
 
-async sub update_accounts {
-    my $self = shift;
-    while (1) {
-        await $self->loop->delay_future(after => UPDATE_ACCOUNTS);
-        $self->{accounts} = await $self->rpc_client->accounts();
-    }
+sub latest_accounts_update : method {
```

Comment:
`:method` while harmless is also completely redundant here. Its only purpose would be to quiet the warnings which can happen when methods take the same name as existing core perl ops (e.g. `shift`). This method name doesn't collide.